articles/cross_compilation_build_tools_design.md

```diff
@@ -0,0 +1,266 @@
+  launching a container a running `docker container export` on it, to a
+  `sysroot` subdirectory of the cc-root, so it can be used as a sysroot during
+  the build.
+  - Those ROS 2 base images are variants of
+    [sysroot/Dockerfile_ubuntu_arm](https://github.com/ros2/cross_compile/blob/master/sysroot/Dockerfile_ubuntu_arm)
+  from ros2/cross_compile.
+  - OSRF would publish those base ROS 2 images, with ROS 2 pre built from
+    source.
+    OSRF would also publish other base ROS 2 images that install
+    `ros-$ROS_DISTRO-desktop` with Apt, instead of installing the system
+    dependencies for ROS 2 with `rosdep install`.
+- The colcon mixin then sets the variable `SYSROOT` to the sysroot path,
+  and other variables required to configure the toolchain file [generic_linux.cmake](https://github.com/ros2/cross_compile/blob/master/cmake-toolchains/generic_linux.cmake)
+  from ros2/cross_compile, and configures `DCMAKE_TOOLCHAIN_FILE` in
+  `cmake-args` pointing to that toolchain file.
+
+The `cc-setup-sysroot` command will also support the following optional arguments:
+
+- `--sysroot-base-image`: Specifies a Docker image that will be used as the
+  base image for the workspace dependent sysroot image. This should be a string
+  that is a valid argument for `docker image pull`, and that can be used in
+  `FROM` Dockerfile statement.
+  If it is not specified, the command will deduce
+  the value of `--sysroot-base-image` from the target platform, for platforms
+  that have an image published for them on [osrf/ros2 on Docker Hub](https://hub.docker.com/r/osrf/ros2/).
+- `--sysroot-image`: Specifies a Docker image for the workspace sysroot. The
+  build will export the sysroot from that image, instead of creating a sysroot
+  image from the workspace.
+  No check is performed to ensure the image has all the
+  dependencies required by the packages in the workspace.
+- `--use-base-image-sysroot`: Boolean flag, when true the base ROS 2 Docker
+  image is used to build the sysroot, so the dependencies declared in packages in
+  the current workspace are ignored.
+  This is useful to speed up builds for
+  packages without additional dependencies.
+
+Support for other platforms can be added by creating corresponding ROS 2 Docker
+images, and variants of the toolchain file if required.
+We could include support for generic versions of ARM-HF and ARM64, on Ubuntu
+Bionic with ROS 2 crystal and fastrtps, in the first release of the commands.
+
+### Frequently asked questions
+
+- _How can we run the resulting binaries?_ Packaging the resulting binaries is
+  outside of the scope of this document.
+  The cross-compiled binaries will be
+  available in the workspace in the cc-root (e.g.
+  `generic_armhf-ubuntu_bionic-fastrtps-crystal`).
+  We make the assumption that
+  the user will copy the whole workspace to the remote device (including `src`)
+  and run `rosdep` on the target platform, to install the dependencies before
+  running the binaries.
```

Comment:
Just a note: this will also install all build dependencies which would't be necessary.